deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -257,62 +250,190 @@ def _copy_all_manifest_files_from_old_hash_buckets(
     return materialize_result_list
 
 
-def _compact_tables(
-    input: MergeInput, dfe_list: List[List[DeltaFileEnvelope]], hb_idx: int
-) -> Tuple[pa.Table, int, int]:
+def _has_previous_compacted_table(input: MergeInput, hb_idx: int) -> bool:
+    """
+    Checks if the given hash bucket index has a compacted table available from the previous compaction round.
+
+    Args:
+        input (MergeInput): The input for the merge operation.
+        hb_idx (int): The hash bucket index to check.
+
+    Returns:
+        bool: True if the hash bucket index has a compacted table available, False otherwise.
+    """
+    return (
+        input.round_completion_info
+        and input.round_completion_info.hb_index_to_entry_range
+        and input.round_completion_info.hb_index_to_entry_range.get(str(hb_idx))
+        is not None
+    )
+
+
+def _flatten_dfe_list(
+    df_envelopes_list: List[List[DeltaFileEnvelope]],
+) -> List[DeltaFileEnvelope]:
+    """
+    Flattens a list of lists of DeltaFileEnvelope objects into a single list of DeltaFileEnvelope objects.
+
+    Args:
+        df_envelopes_list (List[List[DeltaFileEnvelope]]): A list of lists of DeltaFileEnvelope objects.
+
+    Returns:
+        List[DeltaFileEnvelope]: A flattened list of DeltaFileEnvelope objects.
+    """
+    if not df_envelopes_list:
+        return []
+    return [d for dfe_list in df_envelopes_list for d in dfe_list]
+
+
+def _sort_df_envelopes(
+    df_envelopes: List[DeltaFileEnvelope],
+    key: Callable = lambda df: (df.stream_position, df.file_index),
+) -> List[DeltaFileEnvelope]:
+    """
+    Sorts a list of DeltaFileEnvelope objects based on a specified key function.
+
+    Args:
+        df_envelopes (List[DeltaFileEnvelope]): A list of DeltaFileEnvelope objects.
+        key (Callable, optional): A function that takes a DeltaFileEnvelope object and returns a key for sorting.
+            Defaults to lambda df: (df.stream_position, df.file_index).
+
+    Returns:
+        List[DeltaFileEnvelope]: A sorted list of DeltaFileEnvelope objects.
+    """
+    if not df_envelopes:
+        return []
+    return sorted(
+        df_envelopes,
+        key=key,
+        reverse=False,  # ascending
+    )
+
+
+def _group_sequence_by_delta_type(
+    df_envelopes: List[DeltaFileEnvelope],
+) -> Iterator[Tuple[List, List]]:
+    """
+    Groups a list of DeltaFileEnvelope objects by their delta_type.
+
+    Args:
+        df_envelopes (List[DeltaFileEnvelope]): A list of DeltaFileEnvelope objects.
+
+    Yields:
+        Iterator[Tuple[DeltaType, List[DeltaFileEnvelope]]]: A tuple containing the delta_type
+        and a list of DeltaFileEnvelope objects that share the same delta_type.
+    """
+    iter_df_envelopes = iter(df_envelopes)
+    for delta_type, delta_type_sequence in itertools.groupby(
+        iter_df_envelopes, lambda x: x.delta_type
+    ):
+        yield delta_type, list(delta_type_sequence)
+
+
+def _compact_table_v2(
```

Comment:
Minor: I think the naming can be misleading since this module is already under the `compactor_v2` package.